src/Features/Core/Portable/Diagnostics/EngineV1/DiagnosticAnalyzerDriver.cs

```diff
@@ -128,14 +128,14 @@ private CompilationWithAnalyzers GetCompilationWithAnalyzers(Compilation compila
             {
                 Interlocked.CompareExchange(
                     ref _lazyCompilationWithAnalyzers,
-                    _owner.GetCompilationWithAnalyzers(_project, compilation, _concurrentAnalysis, _reportSuppressedDiagnostics),
+                    _owner.GetCompilationWithAnalyzers(_project, analyzers, compilation, _concurrentAnalysis, _reportSuppressedDiagnostics),
                     null);
             }
 
             return _lazyCompilationWithAnalyzers;
         }
 
-        public async Task<ImmutableArray<Diagnostic>> GetSyntaxDiagnosticsAsync(DiagnosticAnalyzer analyzer)
+        public async Task<ImmutableArray<Diagnostic>> GetSyntaxDiagnosticsAsync(IEnumerable<DiagnosticAnalyzer> analyzers, DiagnosticAnalyzer analyzer)
```

Comment:
This API also looks weird, it is unclear if diagnostics are being computed for all analyzers or single analyzer. Passing analyzers in the constructor seems better.